### PR TITLE
Export table API + auto detect data

### PR DESCRIPTION
**Pull request by mabezen**
In this PR we've introduced a new API for inserting a simple table (without power query linked) with an initial data.

One more improvement is introducing a new data type (auto detect) that formats the data according to the cell data content.

- Please don't review code styling - next PR we will add prettify to the project.
- Currently, we are not supporting date type, we are treating dates as strings.
- Next PR will update the documentation before release.

### CODE REVIEW

File: src/utils/documentUtils.ts [hunk 1]
```diff
@@ -79,4 +79,21 @@ const updateCellData = (dataType: DataTypes, data: string, cell: Element, cellDa
...
+const resolveType = (originalDataType: DataTypes, originalData: string | number | boolean) : DataTypes => {
+    if (originalDataType !== DataTypes.autodetect)
+    {
+        return originalDataType;
+    }
+    
+    const data: string = originalData as string
+    let dataType : DataTypes = isNaN(Number(data)) ? DataTypes.string : DataTypes.number;
+    if (dataType == DataTypes.string) {
+       if (data.toLowerCase() == trueStr || data.toLowerCase() == fasleStr) {
```
Comment (piechenick) [inline src/utils/documentUtils.ts:91]: typo: "fasleStr" > "falseStr"

File: src/utils/tableUtils.ts [hunk 2]
```diff
@@ -18,32 +18,35 @@ const updateTableInitialDataIfNeeded = async (zip: JSZip, tableData?: TableData)
...
-            throw new Error(queryTableNotFoundErr);
-        }
+        if (updateQueryTable) {
+            const queryTableXmlString: string | undefined = await zip.file(queryTableXmlPath)?.async(textResultType);
+            if (queryTableXmlString === undefined) {
+                throw new Error(queryTableNotFoundErr);
+            }
+
+            const newQueryTable: string = await updateQueryTablesInitialData(queryTableXmlString, tableData);
+            zip.file(queryTableXmlPath, newQueryTable);
```
Comment (piechenick) [inline src/utils/tableUtils.ts:28]: Why not use await ....async api as above?

File: src/utils/tableUtils.ts [hunk 3]
```diff
@@ -18,32 +18,35 @@ const updateTableInitialDataIfNeeded = async (zip: JSZip, tableData?: TableData)
...
-        const newQueryTable: string = await updateQueryTablesInitialData(queryTableXmlString, tableData);
-        zip.file(queryTableXmlPath, newQueryTable);
+            // update defined name
+            const workbookXmlString: string | undefined = await zip.file(workbookXmlPath)?.async(textResultType);
+            if (workbookXmlString === undefined) {
+                throw new Error(sheetsNotFoundErr);
+            }
+
+            const newWorkbook: string = await updateWorkbookInitialData(workbookXmlString, tableData);
+            zip.file(workbookXmlPath, newWorkbook);
```
Comment (piechenick) [inline src/utils/tableUtils.ts:37]: await.... async api as above?

File: src/utils/tableUtils.ts [hunk 4]
```diff
@@ -18,32 +18,35 @@ const updateTableInitialDataIfNeeded = async (zip: JSZip, tableData?: TableData)
...
+        }
 
         const tableXmlString: string | undefined = await zip.file(tableXmlPath)?.async(textResultType);
         if (tableXmlString === undefined) {
             throw new Error(tableNotFoundErr);
         }
 
-        const newTable: string = await updateTablesInitialData(tableXmlString, tableData);
+        const newTable: string = await updateTablesInitialData(tableXmlString, tableData, updateQueryTable);
         zip.file(tableXmlPath, newTable);
```
Comment (piechenick) [inline src/utils/tableUtils.ts:46]: await... async api as above?

File: src/utils/tableUtils.ts [hunk 5]
```diff
@@ -52,11 +55,14 @@ const updateTablesInitialData = async (tableXmlString: string, tableData: TableD
...
             tableColumn.setAttribute(elementAttributes.id, (columnIndex + 1).toString());
-            tableColumn.setAttribute(elementAttributes.uniqueName, (columnIndex + 1).toString());
             tableColumn.setAttribute(elementAttributes.name, col.name);
-            tableColumn.setAttribute(elementAttributes.queryTableFieldId, (columnIndex + 1).toString());
             tableColumns.appendChild(tableColumn);
             tableColumn.setAttribute(elementAttributes.xr3uid, "{" + v4().toUpperCase() + "}");
+
+            if (updateQueryTable) {
+                tableColumn.setAttribute(elementAttributes.uniqueName, (columnIndex + 1).toString());
+                tableColumn.setAttribute(elementAttributes.queryTableFieldId, (columnIndex + 1).toString());
```
Comment (piechenick) [inline src/utils/tableUtils.ts:64]: Is .toString() really needed? isn't the expected  param already string?

File: tests/documentUtils.test.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    test("ResolveType date not supported success", async () => {
+        expect(documentUtils.resolveType(DataTypes.autodetect, "5-4-2023 00:00")).toEqual(DataTypes.string);
+    });
+
+    test("ResolveType string success", async () => {
+        expect(documentUtils.resolveType(DataTypes.autodetect, "sTrIng")).toEqual(DataTypes.string);
+    });
+    
+    test("ResolveType boolean success", async () => {
+        expect(documentUtils.resolveType(DataTypes.autodetect, "true")).toEqual(DataTypes.boolean);
```
Comment (piechenick) [inline tests/documentUtils.test.ts:15]: What about " true" for example? do we expect white spaces?

File: tests/documentUtils.test.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        expect(documentUtils.resolveType(DataTypes.autodetect, "True")).toEqual(DataTypes.boolean);
+        expect(documentUtils.resolveType(DataTypes.autodetect, "false")).toEqual(DataTypes.boolean);
+        expect(documentUtils.resolveType(DataTypes.autodetect, "False")).toEqual(DataTypes.boolean); 
+    });
+
+    test("ResolveType number success", async () => {
+        expect(documentUtils.resolveType(DataTypes.autodetect, "100000")).toEqual(DataTypes.number);
+        expect(documentUtils.resolveType(DataTypes.autodetect, "1000.00")).toEqual(DataTypes.number);
```
Comment (piechenick) [inline tests/documentUtils.test.ts:23]: Do we expect 1.123? 1e5?

**Review by yarden95c — DISMISSED** [review]

**Review by GuyBenbenisti — APPROVED** [review]

**Comment by mabezen** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Review by ron-b — APPROVED** [review]

**Pull request merged**